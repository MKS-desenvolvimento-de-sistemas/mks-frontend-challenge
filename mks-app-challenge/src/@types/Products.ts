export type DataApi = {
    brand: string,
    description: string,
    id: number,
    name: string,
    photo: string,
    price: string
}

export interface Product {
    id: number;
    name: string;
    price: number;
    description: string;
    photo: string;

}