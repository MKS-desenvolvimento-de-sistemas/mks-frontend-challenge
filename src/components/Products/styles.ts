import styled from "styled-components";

export const List = styled.ul`
  width: 100%;
  max-width: 938px;
  padding: 18px 0 37.96px 0;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  gap: 18px;
`;
