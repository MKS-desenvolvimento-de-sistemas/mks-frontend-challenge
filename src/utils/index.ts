import { IProductCart } from "../types/api";

export const getQuantityInCart = (cart: IProductCart[]) => {
  if (cart.length === 0) return 0;
  const quantities = cart.map((product) => product.quantity);
  const quantityTotal = quantities.reduce(
    (acc, currentValue) => acc + currentValue
  );

  return quantityTotal;
};

export const getPriceInCart = (cart: IProductCart[]) => {
  if (cart.length === 0) return 0;
  const prices = cart.map(
    (product) => Number(product.price) * product.quantity
  );
  const priceTotal = prices.reduce((acc, currentValue) => acc + currentValue);

  return priceTotal;
};
