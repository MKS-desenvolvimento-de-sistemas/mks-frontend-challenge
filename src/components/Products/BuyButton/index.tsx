import * as Styles from "./styles";
import { FiShoppingBag } from "react-icons/fi";
import { IProduct } from "../../../types/api";

// REDUX IMPORT'S
import { add } from "../../../redux/Cart";
import { useDispatch } from "react-redux";

interface IBuyButtonProps {
  product: IProduct;
}

const BuyButton = ({ product }: IBuyButtonProps) => {
  const dispatch = useDispatch();

  return (
    <Styles.Button
      type="button"
      title="Adicionar ao Carrinho"
      onClick={() => dispatch(add(product))}
    >
      <FiShoppingBag size="16px" />
      Comprar
    </Styles.Button>
  );
};

export default BuyButton;
