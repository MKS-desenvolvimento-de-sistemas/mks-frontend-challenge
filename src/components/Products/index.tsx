import * as Styles from "./styles";
import { IProducts } from "../../types/api";
import ProductCard from "./Card";

const ProductsList = ({ products }: IProducts) => {
  return (
    <Styles.List data-testid="products-list">
      {products?.map((product) => (
        <ProductCard key={product.id} {...product} />
      ))}
    </Styles.List>
  );
};

export default ProductsList;
