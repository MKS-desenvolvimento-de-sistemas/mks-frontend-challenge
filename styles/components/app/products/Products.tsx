import axios from "axios";
import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import Loader from "../loader/Loader";
import Img from "../image/Image";
import BuyIcon from "./buyIcon";
import {Grid, Card, Button, TitlePrice} from '../../../styles';
import { MathNumber } from "../tools/math/MathNumber";
import { productList, productCart } from "../state/actions/productsReducers";


const ProductsList: React.FC = () => {
    const ProductsData = useSelector((state: any) => state?.products?.products?.products);
    const dispatch = useDispatch();

    const fetchProducts = async () => {
        await axios.get("https://mks-challenge-api-frontend.herokuapp.com/api/v1/products?page=1&rows=8&sortBy=id&orderBy=ASC")
        .then((res) => {
            // dispatch(getProductsData(res.data))      
            dispatch(productList(res.data))    
        })
    }

    useEffect(() => {
        fetchProducts()
    },[])
  
    return (
        <>         
            {
                (!ProductsData && ProductsData == undefined) ? 
                    <>
                        <Loader/>
                    </>
                    :
                   <Grid>
                     {ProductsData.map((individualProductData: any) => {
                        return (
                            <Card>  
                                <Img src={individualProductData.photo} width={130} height={138}  />
                                <TitlePrice>
                                    <h1>{individualProductData.name}</h1>
                                    <h2>{MathNumber(individualProductData.price)}</h2>
                                </TitlePrice>                                
                                <h3>Redesigned from scratch and completely revised.</h3>      
                                <Button onClick={() =>                                     
                                    dispatch(
                                        productCart({
                                          id: individualProductData.id,
                                          name: individualProductData.name,
                                          price: individualProductData.price,
                                          photo: individualProductData.photo,
                                          quantity: 1
                                        })                                     
                                    )}>
                                  <BuyIcon />
                                  <p>COMPRAR</p>
                                </Button>                                                         
                            </Card >
                        )
                    })}  
                   </Grid>
            }            
        </>
    )
}

export default ProductsList;