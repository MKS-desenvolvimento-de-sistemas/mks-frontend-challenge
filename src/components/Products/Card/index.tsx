import * as Styles from "./styles";

import { IProduct } from "../../../types/api";
import { Price } from "../../";
import BuyButton from "../BuyButton";

const ProductCard = (props: IProduct) => {
  const { name, description, price, photo } = props;
  const formatterDescription =
    description.length > 63 ? description.slice(0, 64) + "..." : description;

  return (
    <Styles.Card data-testid="card">
      <Styles.CardDetails>
        <Styles.CardImage
          data-testid="card--image"
          imageUrl={photo}
        ></Styles.CardImage>

        <Styles.CardInfo>
          <h4>{name}</h4>
          <Price price={Number(price)} type="product" />
          <p>{formatterDescription}</p>
        </Styles.CardInfo>
      </Styles.CardDetails>

      <BuyButton product={props} />
    </Styles.Card>
  );
};

export default ProductCard;
