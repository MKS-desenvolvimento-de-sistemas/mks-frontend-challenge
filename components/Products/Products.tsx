import Card from "@/components/Card/Card";
import styled from "styled-components";
import { Product } from "@/integrations/mks";

const Container = styled.ul`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: 18px auto 50px auto;
  max-width: 1200px;
  gap: 18px;
`;

export interface ProductsProps {
  products: Product[];
  loading: boolean;
  skeletonsCount?: number;
}

const Products = ({ products, loading, skeletonsCount = 8 }: ProductsProps) => {
  return (
    <Container>
      {loading
        ? Array.from({ length: skeletonsCount }).map((_, i) => (
            <Card.Skeleton key={i} />
          ))
        : products.map((product) => (
            <li key={product.id}>
              <Card {...product}></Card>
            </li>
          ))}
    </Container>
  );
};

export default Products;
