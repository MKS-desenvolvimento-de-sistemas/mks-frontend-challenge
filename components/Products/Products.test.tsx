import { render } from "@testing-library/react";
import Products from "@/components/Products/Products";
import { mockProducts, withTheme } from "../../test/utils";

it("should render skeletons", () => {
  const { container } = render(<Products products={[]} loading />);
  expect(container).toMatchSnapshot();
});

it("should render given products", () => {
  const { container } = render(
    withTheme(<Products products={mockProducts} loading={false} />)
  );
  expect(container).toMatchSnapshot();
});
