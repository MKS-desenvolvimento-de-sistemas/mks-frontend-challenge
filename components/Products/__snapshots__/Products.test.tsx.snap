// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render given products 1`] = `
<div>
  <ul
    class="sc-kDDrLX iPcPaR"
  >
    <li>
      <div
        class="sc-gsnTZi syHTx"
      >
        <img
          alt="Iphone 11 128 GB"
          class="sc-dkzDqf bFvDCI"
          src="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/iphone11x128.webp"
        />
        <div
          class="sc-jSMfEi laoEdf"
        >
          <h1
            class="sc-hKMtZM eDkWcw"
          >
            Iphone 11 128 GB
          </h1>
          <span
            class="sc-bczRLJ fvLqnu"
            style="height: 30px;"
          >
            R$ 5.000
          </span>
        </div>
        <p
          class="sc-eCYdqJ dfRMpl"
        >
          Grave vídeos 4K, faça belos retratos e capture paisagens inteiras com o novo sistema de câmera dupla.
        </p>
        <div
          style="flex: 1;"
        />
        <button
          class="sc-gKXOVf hEQAyR"
        >
          <svg
            class="sc-iBkjds dFEFZF"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <path
              d="M16 10a4 4 0 0 1-8 0"
            />
          </svg>
           Comprar
        </button>
      </div>
    </li>
    <li>
      <div
        class="sc-gsnTZi syHTx"
      >
        <img
          alt="AirPods"
          class="sc-dkzDqf bFvDCI"
          src="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/airpods.webp"
        />
        <div
          class="sc-jSMfEi laoEdf"
        >
          <h1
            class="sc-hKMtZM eDkWcw"
          >
            AirPods
          </h1>
          <span
            class="sc-bczRLJ fvLqnu"
            style="height: 30px;"
          >
            R$ 1.200
          </span>
        </div>
        <p
          class="sc-eCYdqJ dfRMpl"
        >
          Criados pela Apple Ligam e se conectam automaticamente Configuração com apenas um toque para todos seus aparelhos Apple.
        </p>
        <div
          style="flex: 1;"
        />
        <button
          class="sc-gKXOVf hEQAyR"
        >
          <svg
            class="sc-iBkjds dFEFZF"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <path
              d="M16 10a4 4 0 0 1-8 0"
            />
          </svg>
           Comprar
        </button>
      </div>
    </li>
    <li>
      <div
        class="sc-gsnTZi syHTx"
      >
        <img
          alt="Macbook Air"
          class="sc-dkzDqf bFvDCI"
          src="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/macbookair.webp"
        />
        <div
          class="sc-jSMfEi laoEdf"
        >
          <h1
            class="sc-hKMtZM eDkWcw"
          >
            Macbook Air
          </h1>
          <span
            class="sc-bczRLJ fvLqnu"
            style="height: 30px;"
          >
            R$ 8.200
          </span>
        </div>
        <p
          class="sc-eCYdqJ dfRMpl"
        >
          Processador intel Core i5 de dois núcleos e 1,8 GHz (Turbo Boost de até 2,9 GHz) com cache L3 compartilhado de 3 MB.
        </p>
        <div
          style="flex: 1;"
        />
        <button
          class="sc-gKXOVf hEQAyR"
        >
          <svg
            class="sc-iBkjds dFEFZF"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <path
              d="M16 10a4 4 0 0 1-8 0"
            />
          </svg>
           Comprar
        </button>
      </div>
    </li>
    <li>
      <div
        class="sc-gsnTZi syHTx"
      >
        <img
          alt="iPhone 12 64 GB"
          class="sc-dkzDqf bFvDCI"
          src="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/iphone12x64.webp"
        />
        <div
          class="sc-jSMfEi laoEdf"
        >
          <h1
            class="sc-hKMtZM eDkWcw"
          >
            iPhone 12 64 GB
          </h1>
          <span
            class="sc-bczRLJ fvLqnu"
            style="height: 30px;"
          >
            R$ 6.500
          </span>
        </div>
        <p
          class="sc-eCYdqJ dfRMpl"
        >
          Grave vídeos 4K, faça belos retratos e capture paisagens inteiras com o novo sistema de câmera dupla.
        </p>
        <div
          style="flex: 1;"
        />
        <button
          class="sc-gKXOVf hEQAyR"
        >
          <svg
            class="sc-iBkjds dFEFZF"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <path
              d="M16 10a4 4 0 0 1-8 0"
            />
          </svg>
           Comprar
        </button>
      </div>
    </li>
    <li>
      <div
        class="sc-gsnTZi syHTx"
      >
        <img
          alt="Apple Watch Series 7"
          class="sc-dkzDqf bFvDCI"
          src="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/applewatch-series7.webp"
        />
        <div
          class="sc-jSMfEi laoEdf"
        >
          <h1
            class="sc-hKMtZM eDkWcw"
          >
            Apple Watch Series 7
          </h1>
          <span
            class="sc-bczRLJ fvLqnu"
            style="height: 30px;"
          >
            R$ 3.200
          </span>
        </div>
        <p
          class="sc-eCYdqJ dfRMpl"
        >
          O Apple Watch faz coisas que outros aparelhos não conseguem porque ele fica no seu pulso.
        </p>
        <div
          style="flex: 1;"
        />
        <button
          class="sc-gKXOVf hEQAyR"
        >
          <svg
            class="sc-iBkjds dFEFZF"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <path
              d="M16 10a4 4 0 0 1-8 0"
            />
          </svg>
           Comprar
        </button>
      </div>
    </li>
    <li>
      <div
        class="sc-gsnTZi syHTx"
      >
        <img
          alt="iPad"
          class="sc-dkzDqf bFvDCI"
          src="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/ipadair.webp"
        />
        <div
          class="sc-jSMfEi laoEdf"
        >
          <h1
            class="sc-hKMtZM eDkWcw"
          >
            iPad
          </h1>
          <span
            class="sc-bczRLJ fvLqnu"
            style="height: 30px;"
          >
            R$ 4.200
          </span>
        </div>
        <p
          class="sc-eCYdqJ dfRMpl"
        >
          iPad é uma linha de tablets projetada, desenvolvida e comercializada pela Apple, que funciona com o sistema operacional móvel iOS e iPadOS.
        </p>
        <div
          style="flex: 1;"
        />
        <button
          class="sc-gKXOVf hEQAyR"
        >
          <svg
            class="sc-iBkjds dFEFZF"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <path
              d="M16 10a4 4 0 0 1-8 0"
            />
          </svg>
           Comprar
        </button>
      </div>
    </li>
    <li>
      <div
        class="sc-gsnTZi syHTx"
      >
        <img
          alt="Headset Cloud Revolver"
          class="sc-dkzDqf bFvDCI"
          src="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/hyperx-cloudrevolver.webp"
        />
        <div
          class="sc-jSMfEi laoEdf"
        >
          <h1
            class="sc-hKMtZM eDkWcw"
          >
            Headset Cloud Revolver
          </h1>
          <span
            class="sc-bczRLJ fvLqnu"
            style="height: 30px;"
          >
            R$ 1.000
          </span>
        </div>
        <p
          class="sc-eCYdqJ dfRMpl"
        >
          A linha HyperX Cloud Revolver foi projetada para atender as exigências dos gamers de PC ou de console.
        </p>
        <div
          style="flex: 1;"
        />
        <button
          class="sc-gKXOVf hEQAyR"
        >
          <svg
            class="sc-iBkjds dFEFZF"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <path
              d="M16 10a4 4 0 0 1-8 0"
            />
          </svg>
           Comprar
        </button>
      </div>
    </li>
    <li>
      <div
        class="sc-gsnTZi syHTx"
      >
        <img
          alt="Headset Cloud Stinger"
          class="sc-dkzDqf bFvDCI"
          src="https://mks-sistemas.nyc3.digitaloceanspaces.com/products/hyperxcloudstinger.webp"
        />
        <div
          class="sc-jSMfEi laoEdf"
        >
          <h1
            class="sc-hKMtZM eDkWcw"
          >
            Headset Cloud Stinger
          </h1>
          <span
            class="sc-bczRLJ fvLqnu"
            style="height: 30px;"
          >
            R$ 600
          </span>
        </div>
        <p
          class="sc-eCYdqJ dfRMpl"
        >
          O HyperX Cloud Stinger™ é o headset ideal para jogadores que procuram leveza e conforto, qualidade de som superior e maior praticidade.
        </p>
        <div
          style="flex: 1;"
        />
        <button
          class="sc-gKXOVf hEQAyR"
        >
          <svg
            class="sc-iBkjds dFEFZF"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <path
              d="M16 10a4 4 0 0 1-8 0"
            />
          </svg>
           Comprar
        </button>
      </div>
    </li>
  </ul>
</div>
`;

exports[`should render skeletons 1`] = `
<div>
  <ul
    class="sc-kDDrLX iPcPaR"
  >
    <div
      class="sc-gsnTZi sc-jqUVSM syHTx ccbHaF"
    >
      <div
        class="sc-ftvSup diTfsO"
      />
      <div
        class="sc-papXJ fekGrp"
      />
      <div
        class="sc-papXJ fekGrp"
      />
      <div
        class="sc-papXJ fekGrp"
      />
    </div>
    <div
      class="sc-gsnTZi sc-jqUVSM syHTx ccbHaF"
    >
      <div
        class="sc-ftvSup diTfsO"
      />
      <div
        class="sc-papXJ fekGrp"
      />
      <div
        class="sc-papXJ fekGrp"
      />
      <div
        class="sc-papXJ fekGrp"
      />
    </div>
    <div
      class="sc-gsnTZi sc-jqUVSM syHTx ccbHaF"
    >
      <div
        class="sc-ftvSup diTfsO"
      />
      <div
        class="sc-papXJ fekGrp"
      />
      <div
        class="sc-papXJ fekGrp"
      />
      <div
        class="sc-papXJ fekGrp"
      />
    </div>
    <div
      class="sc-gsnTZi sc-jqUVSM syHTx ccbHaF"
    >
      <div
        class="sc-ftvSup diTfsO"
      />
      <div
        class="sc-papXJ fekGrp"
      />
      <div
        class="sc-papXJ fekGrp"
      />
      <div
        class="sc-papXJ fekGrp"
      />
    </div>
    <div
      class="sc-gsnTZi sc-jqUVSM syHTx ccbHaF"
    >
      <div
        class="sc-ftvSup diTfsO"
      />
      <div
        class="sc-papXJ fekGrp"
      />
      <div
        class="sc-papXJ fekGrp"
      />
      <div
        class="sc-papXJ fekGrp"
      />
    </div>
    <div
      class="sc-gsnTZi sc-jqUVSM syHTx ccbHaF"
    >
      <div
        class="sc-ftvSup diTfsO"
      />
      <div
        class="sc-papXJ fekGrp"
      />
      <div
        class="sc-papXJ fekGrp"
      />
      <div
        class="sc-papXJ fekGrp"
      />
    </div>
    <div
      class="sc-gsnTZi sc-jqUVSM syHTx ccbHaF"
    >
      <div
        class="sc-ftvSup diTfsO"
      />
      <div
        class="sc-papXJ fekGrp"
      />
      <div
        class="sc-papXJ fekGrp"
      />
      <div
        class="sc-papXJ fekGrp"
      />
    </div>
    <div
      class="sc-gsnTZi sc-jqUVSM syHTx ccbHaF"
    >
      <div
        class="sc-ftvSup diTfsO"
      />
      <div
        class="sc-papXJ fekGrp"
      />
      <div
        class="sc-papXJ fekGrp"
      />
      <div
        class="sc-papXJ fekGrp"
      />
    </div>
  </ul>
</div>
`;
